### PR TITLE
Use new templates for cilium 1.8

### PR DESCRIPTION
**Pull request by olemarkus**
WIP at least until cilium 1.8 has been released.

Using a new template here since cilium templates have changed quite a bit for 1.8.

Also adding support for hubble. Right now, it is just possible to enable it. But need to add support for metrics at least. Not 100% decided if hubble-relay should be part of the addon or not. Possibly leave it to the user to install relay (and ui)

### CODE REVIEW

File: pkg/apis/kops/v1alpha2/networking.go [hunk 1]
```diff
@@ -446,6 +448,12 @@ type CiliumNetworkingSpec struct {
 	CniBinPath string `json:"cniBinPath,omitempty"`
 }
 
+// Hubble configures Hubble service on the Cilium agent
```
Comment (johngmyers) [inline pkg/apis/kops/v1alpha2/networking.go:451]:
```suggestion
// HubbleSpec configures the Hubble service on the Cilium agent.
```

File: pkg/apis/kops/v1alpha2/networking.go [hunk 2]
```diff
@@ -446,6 +448,12 @@ type CiliumNetworkingSpec struct {
...
+type HubbleSpec struct {
+	// Enabled decides if Hubble is enabled on the agent or not
```
Comment (johngmyers) [inline pkg/apis/kops/v1alpha2/networking.go:453]:
```suggestion
	// Enabled specifies whether Hubble is enabled on the agent.
```

File: pkg/wellknownports/wellknownports.go [hunk 3]
```diff
@@ -51,6 +51,12 @@ const (
 
 	// NodeLocalDNSHealthCheck is the port where the node-local-dns health check listens.
 	NodeLocalDNSHealthCheck = 3989
+
+	// CiliumPrometheusPort is the default port where Cilium exposes metrics
+	CiliumPrometheusPort = 9090
+
+	// CiliumHubblePrometheusPort is the default port where Hubble exposes metrics
+	CiliumHubblePrometheusPort = 9091
```
[the remark this answers is not in the document]
Comment (olemarkus) [reply pkg/wellknownports/wellknownports.go:59]: Yep. These ports are exposed on the CNI agent itself.

File: pkg/model/components/cilium.go [hunk 4]
```diff
@@ -39,7 +40,7 @@ func (b *CiliumOptionsBuilder) BuildOptions(o interface{}) error {
 		if b.Context.IsKubernetesLT("1.12.0") {
 			c.Version = "v1.6.9"
 		} else {
-			c.Version = "v1.7.4"
+			c.Version = "v1.8.0-rc4"
```
[the remark this answers is not in the document]
Comment (olemarkus) [reply pkg/model/components/cilium.go:43]: We should now that 1.8.0 is actually released. Will fix.

Normally we don't do this with CNIs unless there are manifest incompatibilities, but the code changes I've made makes this an option. I am neutral on this.

File: upup/pkg/fi/cloudup/tests/bootstrapchannelbuilder/cilium/manifest.yaml [hunk 5]
```diff
@@ -86,19 +86,11 @@ spec:
...
-    manifestHash: 590dd7dc770d721f15d63e8983aa253010757ddd
-    name: networking.cilium.io
-    selector:
-      role.kubernetes.io/networking: "1"
-    version: 1.7.3-kops.1
   - id: k8s-1.12
     kubernetesVersion: '>=1.12.0'
-    manifest: networking.cilium.io/k8s-1.12.yaml
-    manifestHash: 61f05c6e376a570b3f1e53d6b0b2ed9e63cf4c50
+    manifest: networking.cilium.io/k8s-1.12-v1.8.yaml
```
[the remark this answers is not in the document]
Comment (olemarkus) [reply upup/pkg/fi/cloudup/tests/bootstrapchannelbuilder/cilium/manifest.yaml:91]: Yep. That makes sense.

File: pkg/apis/kops/validation/validation.go [hunk 6]
```diff
@@ -657,24 +658,37 @@ func validateNetworkingCilium(cluster *kops.Cluster, v *kops.CiliumNetworkingSpe
...
+		if version.Minor == 6 && (!cluster.IsKubernetesGTE("1.11") || cluster.IsKubernetesGTE("1.16")) {
+			allErrs = append(allErrs, field.Forbidden(versionFld, "Version 1.6 requires kubernetesVersion between 1.11 and 1.16"))
+		}
 
-		if !(version.GTE(v6) && version.LT(v8)) {
-			allErrs = append(allErrs, field.Invalid(versionFld, v.Version, "Only versions 1.6 and 1.7 are supported"))
+		if version.Minor == 7 && (!cluster.IsKubernetesGTE("1.12") || cluster.IsKubernetesGTE("1.17")) {
+			allErrs = append(allErrs, field.Forbidden(versionFld, "Version 1.7 requires kubernetesVersion between 1.12 and 1.17"))
+		}
+		if version.Minor == 8 && (!cluster.IsKubernetesGTE("1.12") || (cluster.IsKubernetesGTE("1.18") && os.Getenv("KOPS_RUN_TOO_NEW_VERSION") == "")) {
```
Comment (hakman) [inline pkg/apis/kops/validation/validation.go:684]: I doubt this won't work once Kubernetes 1.19 is released.
Also think about using or adding `cluster.IsKubernetesLT`.
```suggestion
		if version.Minor == 8 && !cluster.IsKubernetesGTE("1.12") {
```

Comment (olemarkus) [reply pkg/apis/kops/validation/validation.go:684]: You are probably correct. I also see that the cilium docs have changed the wording slightly: https://docs.cilium.io/en/v1.8/concepts/kubernetes/requirements/#kubernetes-version

File: pkg/apis/kops/validation/validation.go [hunk 7]
```diff
@@ -657,24 +658,37 @@ func validateNetworkingCilium(cluster *kops.Cluster, v *kops.CiliumNetworkingSpe
...
+			allErrs = append(allErrs, field.Forbidden(versionFld, "Version 1.6 requires kubernetesVersion between 1.11 and 1.16"))
+		}
 
-		if !(version.GTE(v6) && version.LT(v8)) {
-			allErrs = append(allErrs, field.Invalid(versionFld, v.Version, "Only versions 1.6 and 1.7 are supported"))
+		if version.Minor == 7 && (!cluster.IsKubernetesGTE("1.12") || cluster.IsKubernetesGTE("1.17")) {
+			allErrs = append(allErrs, field.Forbidden(versionFld, "Version 1.7 requires kubernetesVersion between 1.12 and 1.17"))
+		}
+		if version.Minor == 8 && !cluster.IsKubernetesGTE("1.12") {
+			allErrs = append(allErrs, field.Forbidden(versionFld, "Version 1.8 requires kubernetesVersion between 1.12 and 1.18"))
```
Comment (hakman) [inline pkg/apis/kops/validation/validation.go:685]:
```suggestion
			allErrs = append(allErrs, field.Forbidden(versionFld, "Version 1.8 requires kubernetesVersion 1.12 or newer"))
```

**Comment by olemarkus** [comment]
Hm. Nodeup is slower than usual.

/retest

**Comment by hakman** [comment]
@johngmyers any other thoughts here, or it's good from your point of view also?

**Comment by johngmyers** [comment]
Good from my point of view.

**Comment by rifelpet** [comment]
/lgtm cancel

might need a rebase + hack/verify-* scripts to fix the failing jobs

**Pull request merged**